### PR TITLE
Task/21716 dashboard search config

### PR DESCRIPTION
**Pull request by ozelenyi**
https://tracker.jbs.com.ua/issues/21716

### CODE REVIEW

File: src/app/application/application.component.ts [hunk 1]
```diff
@@ -155,6 +166,7 @@ export class ApplicationComponent implements OnInit, OnDestroy {
     }
 
     protected buildOptions(defaultFields) {
+        console.log(this.searchParams);
```
[the remark this answers is not in the document]
Comment (major13ua) [reply src/app/application/application.component.ts]: Please write logs in Dev mode. There are lots of examples how to to it in codebase.

File: src/app/application/application.component.ts [hunk 2]
```diff
@@ -261,4 +270,18 @@ export class ApplicationComponent implements OnInit, OnDestroy {
         return spec.types.filter(t => t.key === typeKey).shift();
     }
 
+    private getSearchConfig(idOrSlug: any): Promise<any> {
```
Comment (major13ua) [inline src/app/application/application.component.ts]: Same code is written in navbar nomponent. Please refactor.

File: src/app/layouts/navbar/navbar.component.ts [hunk 3]
```diff
@@ -228,6 +231,22 @@ export class NavbarComponent implements OnInit, OnDestroy, DoCheck {
         this.location.back();
     }
 
+    public getSearchMask(): void {
+        this.dashboardWrapperService.dashboards()
```
Comment (major13ua) [inline src/app/layouts/navbar/navbar.component.ts]: Same code in application component. Please refactor.

Comment (ozelenyi) [reply src/app/layouts/navbar/navbar.component.ts]: Navbar and application components are connected via router. Within navbar component I need dashboard config for search input element mask. Also I need config in application component for displaying results. The thing is I can`t pass config object from navbar to application component via router query params. Maybe I could use event manager. What solution can you suggest?

File: src/app/xm-dashboard/shared/dashboard-wrapper.service.ts [hunk 4]
```diff
@@ -54,4 +56,25 @@ export class DashboardWrapperService {
         }
     }
 
+    public getDashboardByIdOrSlug(idOrSlug: number | string, force?: boolean): Observable<Dashboard> {
+        if (!force && this._dashboards && this._dashboards.length) {
+            return of(this._dashboards
```
Comment (major13ua) [inline src/app/xm-dashboard/shared/dashboard-wrapper.service.ts]: строка 61 и 70 - дублирование

**Pull request merged**